From e38277b272d1bc015d0b64da88782a8506dfb7f9 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Wed, 19 Dec 2018 16:43:30 +0530
Subject: [PATCH 0494/1852] dt: bindings: media: xilinx: Add clock support to
 AXI4 Stream switch

commit 6f67571002e885d13a7cd6e02540da69e196a0ce from
https://github.com/Xilinx/linux-xlnx.git

This patch adds support for clock framework in the AXI4 Stream switch.
Two clock properties are added for video clock and axi lite interface
clock. The axi lite interface clock is required to be present when
control register routing method is enabled i.e. routing mode is 1.

The patch breaks backward compatibility as the IP version has remained
same but clocks are new required properties.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-axi4s-switch.txt        | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-axi4s-switch.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-axi4s-switch.txt
index 6ae1c8ed0fd3..fb5ed47d959a 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-axi4s-switch.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-axi4s-switch.txt
@@ -27,10 +27,15 @@ Required properties:
  - xlnx,num-si-slots: Number of slave / input ports. Min 1 Max 16 .
  - xlnx,num-mi-slots: Number of master / output ports. Min 1 Max 16.
  - ports: Video ports, using the DT bindings defined in ../video-interfaces.txt.
+ - clocks: Reference to the AXI Streaming clock feeding the ACLK and
+   AXI4 Lite control interface clock when control routing is enabled.
+ - clock-names: Must have "aclk".
 
 Optional properties:
  - reg: Physical base address and length of the registers set for the device.
 	This is required only if xlnx,routing-mode is 1.
+ - clocks: Reference to AXI4 Lite control interface clock when routing-mode is 1.
+ - clock-names: "s_axi_ctl_clk" clock for AXI4 Lite interface when routing-mode is 1.
 
 Example:
 
@@ -41,6 +46,8 @@ For TDEST routing, from 1 slave port to 4 master ports
 		xlnx,routing-mode = <0x0>;
 		xlnx,num-si-slots = <0x1>;
 		xlnx,num-mi-slots = <0x4>;
+		clocks = <&vid_stream_clk>;
+		clock-names = "aclk";
 
 		ports {
 			#address-cells = <1>;
@@ -87,6 +94,8 @@ For Control reg based routing, from 2 slave ports to 4 master ports
 		xlnx,routing-mode = <0x1>;
 		xlnx,num-si-slots = <0x2>;
 		xlnx,num-mi-slots = <0x4>;
+		clocks = <&vid_stream_clk>, <&misc_clk_0>;
+		clock-names = "aclk", "s_axi_ctl_clk;
 
 		ports {
 			#address-cells = <1>;
-- 
2.31.1

